src/components/Widgets/TableWidget.tsx

```diff
@@ -0,0 +1,40 @@
+import { Box } from '@mui/material';
+import { DataGridPro } from '@mui/x-data-grid-pro';
+import { useRef } from 'react';
+import { GenericWidget } from './GenericWidget';
+import { TableWidgetProps } from './widgetConfig';
+
+export const TableWidget = ({
+  columns,
+  dataGridProps,
+  ...props
+}: TableWidgetProps) => {
+  const containerRef = useRef<HTMLDivElement>(null);
+  const maxHeight = containerRef.current?.clientHeight ?? 300;
+
+  return (
+    <GenericWidget {...props} ref={containerRef}>
+      <Box
+        sx={({ spacing }) => ({
+          height: '100%',
+          width: `calc(100% - ${spacing(4)})`,
+          display: 'flex',
+          flexDirection: 'column',
+          justifyContent: 'center',
+        })}
+      >
+        <DataGridPro
+          density="compact"
+          {...dataGridProps}
+          columns={columns}
+          sx={{
+            '--unstable_DataGrid-radius': '0px',
+            border: 'none',
+            maxHeight,
```

Comment:
Instead of using JS to find element height, you can use a container query.
```tsx
<Box sx={{ containerType: 'size' }}>
  <DataGridPro sx={{ height: '100cqh' }} />
</Box>
```

Reply:
The DataGrid implementation is being moved to the specific Widget (ie: TeamNewsWidget) and will be addressed there.